src/boss/lambdaStore.go

```diff
@@ -0,0 +1,300 @@
+package boss
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-lambda/open-lambda/ol/common"
+)
+
+type LambdaStore struct {
+	StorePath    string
+	Lambdas      []LambdaEntry
+	HTTPTriggers []HTTPEntry
+	CronTriggers []CronEntry
+}
+
+type LambdaEntry struct {
+	FunctionName string
+	Config       *common.LambdaConfig
+}
+
+type HTTPEntry struct {
+	FunctionName string
+	common.HTTPTrigger
+}
+
+type CronEntry struct {
+	FunctionName string
+	common.CronTrigger
+}
+
+func NewLambdaStore(storePath string) (*LambdaStore, error) {
+	store := &LambdaStore{StorePath: storePath}
+
+	if err := os.MkdirAll(store.StorePath, 0755); err != nil {
+		return nil, fmt.Errorf("failed to create lambda store directory: %w", err)
+	}
+
+	files, _ := os.ReadDir(store.StorePath)
+	for _, file := range files {
+		if file.IsDir() {
+			funcName := file.Name()
+			if err := store.loadConfigAndRegister(funcName); err != nil {
+				log.Printf("Failed to load lambda %s: %v", funcName, err)
+			}
+		}
+	}
+	return store, nil
+}
+
+// ------------------- HTTP Handlers ----------------------
+
+func (s *LambdaStore) UploadLambda(w http.ResponseWriter, r *http.Request) {
+	rawName := strings.TrimPrefix(r.URL.Path, "/lambda/upload/")
+	functionName, err := sanitizeFunctionName(rawName)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	tarPath := filepath.Join(s.StorePath, functionName+".tar.gz")
+	os.Remove(tarPath)
+
+	tarFile, err := os.Create(tarPath)
+	if err != nil {
+		http.Error(w, "Failed to create lambda file", http.StatusInternalServerError)
+		return
+	}
+	if _, err := io.Copy(tarFile, r.Body); err != nil {
+		http.Error(w, "Failed to save uploaded file", http.StatusInternalServerError)
+		tarFile.Close()
+		return
+	}
+	tarFile.Close()
+
+	// Reopen tarball to extract just config.json
+	f, err := os.Open(tarPath)
+	if err != nil {
+		http.Error(w, "Failed to reopen uploaded tarball", http.StatusInternalServerError)
+		return
+	}
+	defer f.Close()
+
+	cfg, err := extractConfigFromTarGz(f)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Failed to extract config.json: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	s.unregisterTriggers(functionName)
+	s.removeFromRegistry(functionName)
+	s.addToRegistry(functionName, cfg)
+	s.registerTriggers(functionName, cfg)
+
+	w.WriteHeader(http.StatusCreated)
+	fmt.Fprintf(w, "Lambda %s uploaded successfully", functionName)
+}
+
+func (s *LambdaStore) DeleteLambda(w http.ResponseWriter, r *http.Request) {
+	raw := strings.TrimPrefix(r.URL.Path, "/lambda/")
```

Comment:
It seems that creating/deleting files for functions should be done in addToRegistry and removeFromRegistry so that it is easy to keep on-disk and in-memory state in sync.  Then somebody writing new code won't have to worry about taking multiple steps to update both.